dial_delay.go

```diff
@@ -0,0 +1,209 @@
+package swarm
+
+import (
+	"context"
+	"time"
+
+	ma "github.com/multiformats/go-multiaddr"
+	mafmt "github.com/whyrusleeping/mafmt"
+)
+
+const p_circuit = 290
+
+const numTiers = 2
+
+var TierDelay = 1 * time.Second
+
+var relay = mafmt.Or(mafmt.And(mafmt.Base(p_circuit), mafmt.Base(ma.P_IPFS)), mafmt.And(mafmt.Base(ma.P_IPFS), mafmt.Base(p_circuit), mafmt.Base(ma.P_IPFS)))
```

Comment:
I wonder if we could have the transports have some way to indicate that they should be delayed, possibly an optional interface like
```
type ExpensiveTransport interface {
  TransportCost() int //basically the 'tier' to use here
}
```

Reply:
We should be aware of potential problems with changes in the circuit addressing here -- cf https://github.com/libp2p/specs/issues/72

Reply:
This might break with the changes in https://github.com/libp2p/go-libp2p-circuit/pull/48
We should drop the trailing /ipfs part from the filter and just rely on the presence of /p2p-circuit

Reply:
Also problematic for explicit relay addresses.